actions/src/openstack_check_actions.py

```diff
@@ -0,0 +1,463 @@
+                            },
+                        }
+                    )
+                    print("Rule not applied to " + server["id"])
+        return rules_with_issues
+
+    def security_groups_check(
+        self,
+        cloud_account: str,
+        max_port: int,
+        min_port: int,
+        ip_prefix: str,
+        project_id=None,
+        all_projects=False,
+    ):
+        """
+        Starts the script to check projects, can choose to go through all projects or a single project.
+
+        Params:
+        Cloud (Required): The name of the cloud to open from clouds.yaml
+
+        Requires one of:
+            all_projects (optional): Boolean for checking all projects, defaults to False
+            project (optional): ID of project to check
+        """
+
+        # all_servers.filter(location.project.id=project)
+
+        # print(servers.get_all_servers())
+        rules_with_issues = {
+            "title": "Server {p[id]} has an incorrectly configured server group",
+            "body": "Project id: {p[proj_id]}\nSecurity Group ID: {p[sec_id]}\n Is applied: {p[applied]}",
+            "server_list": [],
+        }
+
+        if all_projects:
+            with OpenstackConnection(cloud_name=cloud_account) as conn:
+                projects = conn.list_projects()
+            for project in projects:
+                output = self._check_project_loadbalancers(
+                    project=project["id"],
+                    cloud=cloud_account,
+                    max_port=max_port,
+                    min_port=min_port,
+                    ip_prefix=ip_prefix,
+                )
+                rules_with_issues["server_list"].extend(output)
+        else:
+            output = self._check_project_loadbalancers(
+                project=project_id,
+                cloud=cloud_account,
+                max_port=max_port,
+                min_port=min_port,
+                ip_prefix=ip_prefix,
+            )
+            rules_with_issues["server_list"] = output
+
+        return rules_with_issues
+
+    def deleting_machines_check(
+        self, cloud_account: str, project_id=None, all_projects=False
+    ):
+        """
+        Action to check for machines that are stuck deleting for more than 10mins
+        Outputs a suitable dictionary to pass into create_tickets
+        """
+        output = {
+            "title": "Server {p[id]} has not been updated in more than 10 minutes during {p[action]}",
+            "body": "The following information may be useful\nHost id: {p[id]} \n{p[data]}",
+            "server_list": [],
+        }
+
+        if all_projects:
+            with OpenstackConnection(cloud_name=cloud_account) as conn:
+                projects = conn.list_projects()
+            for project in projects:
+                deleted = self._check_deleted(
+                    project=project["id"], cloud=cloud_account
+                )
+            output["server_list"].extend(deleted)
```

Comment:
There's a bug here, only the final iteration of the loop will be appended as it's not indented enough